test/service/s3_service_test.rb

```diff
@@ -30,6 +30,20 @@ class ActiveStorage::Service::S3ServiceTest < ActiveSupport::TestCase
       assert_match /#{SERVICE_CONFIGURATIONS[:s3][:bucket]}\.s3.(\S+)?amazonaws.com.*response-content-disposition=inline.*avatar\.png/,
         @service.url(FIXTURE_KEY, expires_in: 5.minutes, disposition: :inline, filename: "avatar.png")
     end
+
+    test "encrypt file with server_side_encryption upload option" do
+      skip "server_side_encryption option not supplied " unless @service.upload_options[:server_side_encryption]
+
+      begin
+        key  = SecureRandom.base58(24)
+        data = "Something else entirely!"
+        response = @service.upload(key, StringIO.new(data), checksum: Digest::MD5.base64digest(data))
+
+        assert_equal @service.upload_options[:server_side_encryption], response.server_side_encryption
```

Comment:
After thinking about this more, I’m uncomfortable with relying on the return value of `S3Service#upload` in this test, given that it is (or will be) unspecified.

Instead, we can look to the object’s metadata:

```ruby
key  = SecureRandom.base58(24)
data = "Something else entirely!"
@service.upload key, StringIO.new(data), checksum: Digest::MD5.base64digest(data)

assert_equal :aes256, @service.bucket.object(key).server_side_encryption
```